### PR TITLE
AP_RPM: enable AP_RPM to set ESC Telemetry RPM

### PR DESCRIPTION
**Pull request by magicrub**
AP_RPM: enable AP_RPM to set ESC Telemetry RPM

### CODE REVIEW

**Comment by peterbarker** [comment]
Might be nice to say why this is useful :-)

Should this be an option within the existing backend indicating which way data is supposed to flow?

**Comment by magicrub** [comment]
It's the only way to get data from AP_RPM into into the mavlink telemetry data? Also, with https://github.com/ArduPilot/ardupilot/pull/23263 we can now also get RPM data in a Periph be useful. I have a "motor periph" that converts DroneCAN to PWM-out, acts as an RPM feedback monitor, motor and esc temperature sensor, and voltage and current monitor.

**Comment by peterbarker** [comment]
> It's the only way to get data from AP_RPM into into the mavlink telemetry data? Also, with #23263 we can now also get RPM 

That's not true - we've had the `RPM` mavlink message since forever.  ITYM "ESC_TELEMETRY_1_TO_4 and friends mavlink telemetry data", yes?

Further to my "do it in the backend?" query, I'm q bit converned about what happens on a SLT quadplane with RPM on a forward motor but using ESC telemetry to get RPM on the lifting motors.

File: libraries/AP_RPM/AP_RPM.cpp [hunk 1]
```diff
@@ -199,6 +200,16 @@ void AP_RPM::update(void)
             }
 
             drivers[i]->update();
+
+#if HAL_WITH_ESC_TELEM
+            if (_params[i].type != RPM_TYPE_ESC_TELEM && _params[i].esc_mask > 0 && healthy(i)) {
```
Comment (andyp1per) [inline libraries/AP_RPM/AP_RPM.cpp:205]: shouldn't this be == RPM_TYPE_ESC_TELEM

**Review by andyp1per — CHANGES_REQUESTED** [review]
Logic looks wrong to me, or else requires a comment

File: libraries/AP_RPM/AP_RPM.cpp (hunk 1 above)
Comment (magicrub) [reply libraries/AP_RPM/AP_RPM.cpp:205]: no, actually. A comment would be useful there. This is functioning as the opposite of RPM_TYPE_ESC_TELEM

RPM_TYPE_ESC_TELEM = read ESC_TELEM and store it in AP_RPM
RPM_TYPE_EverythingElse = read RPM and write to ESC_TELEM

Comment (magicrub) [reply libraries/AP_RPM/AP_RPM.cpp:208]: @andyp1per I've added a comment.

**Review by andyp1per — APPROVED** [review]

File: libraries/AP_RPM/AP_RPM_Params.cpp [hunk 2]
```diff
@@ -62,8 +62,8 @@ const AP_Param::GroupInfo AP_RPM_Params::var_info[] = {
     AP_GROUPINFO("PIN", 6, AP_RPM_Params, pin, -1),
 
     // @Param: ESC_MASK
-    // @DisplayName: Bitmask of ESC telemetry channels to average
-    // @Description: Mask of channels which support ESC rpm telemetry. RPM telemetry of the selected channels will be averaged
+    // @DisplayName: Bitmask of ESC telemetry channels
```
Comment (rmackay9) [inline libraries/AP_RPM/AP_RPM_Params.cpp:65]: As a side note re naming, we should drop words like "Bitmask" from the Display Name and should start with the subsystem (e.g. "RPM").  The Display Name is the top level name of the parameter shown to the user so it really should be something like, "RPM ESC telemetry channels".

**Comment by AliSeedg** [comment]
> > It's the only way to get data from AP_RPM into into the mavlink telemetry data? Also, with #23263 we can now also get RPM
> 
> That's not true - we've had the `RPM` mavlink message since forever. ITYM "ESC_TELEMETRY_1_TO_4 and friends mavlink telemetry data", yes?
> 
> Further to my "do it in the backend?" query, I'm q bit converned about what happens on a SLT quadplane with RPM on a forward motor but using ESC telemetry to get RPM on the lifting motors.

Can I ask you some questions?

**Pull request merged**